__tests__/main.test.js

```diff
@@ -184,3 +225,56 @@ test("Branch", async () => {
   );
   expect(tags).toEqual(["main"]);
 });
+
+test("Includes pre-releases one", async () => {
+  const scope = nock("https://api.github.com")
+    .get("/repos/owner/repo/tags")
+    .reply(200, [
+      {
+        name: "1.0.0",
+      },
+      {
+        name: "1.1.0",
+      },
+      {
+        name: "1.1.0-0",
+      },
+    ]);
+  const tags = await calculateTags(
+    "TOKEN",
+    "owner",
+    "repo",
+    "refs/tags/1.1.0-1",
+    ""
+  );
+  expect(tags).toEqual(["1.1.0-1", "1.1.0", "1.1", "1", "latest"]);
+  scope.done();
+});
+
+test("Includes pre-releases one with new tag", async () => {
+  const scope = nock("https://api.github.com")
+    .get("/repos/owner/repo/tags")
+    .reply(200, [
+      {
+        name: "1.0.0",
+      },
+      {
+        name: "1.1.0",
+      },
+      {
+        name: "2.0.0",
+      },
+      {
+        name: "1.1.0-0",
+      },
+    ]);
+  const tags = await calculateTags(
+    "TOKEN",
+    "owner",
+    "repo",
+    "refs/tags/1.1.0-1",
```

Comment:
Hi @jburel! :D I very much appreciate this github action and that it has been allowed to migrate to the JupyterHub org! Thank you for your contributions to it!!!

---

The git reference that triggers the build is tested to be a build version later than those available in the tests above. What do you think about adding one for when you have a the tag `1.0.0` and `1.0.0-2` and then want to calculate the tags from `1.0.0-1`?

I expect our logic could be sensitive to malfunctioning in those situations.

Reply:
I assume that in that case, you expect to only have ``1.0.0-1`` as the returned value

Reply:
Yes, do you think that makes sense?

I just spent some time considering this further and made a PR to this PR: https://github.com/jburel/action-major-minor-tag-calculator/pull/1